src/background.js

```diff
@@ -18,11 +18,36 @@ function log(...args) {
 
 let iconData;
 
+function updateButtonForTab(tabId, changeInfo) {
+  if (!changeInfo.url) {
+    return;
+  }
+  const url = changeInfo.url;
+  if (url.startsWith('http:') || url.startsWith('https:') || url.startsWith('file:') ||
+      url.startsWith('ftp:') || url.startsWith('app:')) {
+    browser.browserAction.setIcon({path: 'icons/Bell Icon.svg', tabId: tabId});
+  } else {
+    browser.browserAction.setIcon({path: 'icons/Disabled Bell Icon.svg', tabId: tabId});
+  }
+}
+
 function init() {
   log('init()');
   browser.alarms.onAlarm.addListener(handleWake);
   browser.notifications.onClicked.addListener(handleNotificationClick);
   browser.runtime.onMessage.addListener(handleMessage);
+  browser.tabs.onUpdated.addListener(updateButtonForTab);
+  browser.tabs.onCreated.addListener(tab => {
+    updateButtonForTab(tab.id, {url: tab.url});
+  });
+  browser.tabs.query({}).then(tabs => {
```

Comment:
It does, and we want it the way it is because we need to set the states for the existing tabs, and we don't get a `tabs.onUpdated` even when we switch tabs.

(i.e. If we start with two tabs, "about:home" and "about:addons", load the extension, we want it to be disabled on both.)